### ISSUE TITLE
Natron does not observe disk,cache size limits, crashes when disk space runs out

### ISSUE THREAD

**Issue by Ichthyostega** [MrKepzie/Natron#1572]
### Problem

Natron does not observe cache size limits. And it just crashes, when disk space runs out.

**Expected behavior:** The ability to set an effective limit on the overall disk space used for caching

**Actual behavior:** uses all available free storage and then crashes

* set viewer cache size to 5 GiB
* set disk cache size likewise to 5 GiB
* direct the cache directory to reside on my SSD
* have one disk cache node in tree
* do some playback (which should already populate part of the caching range)
* klick precache (start caching job)
* crash `0d0018d9-590a-fc77-21d1ff9c-36a63c05`
* no further output in terminal. GUI remains there and seems to work, but process (shell script) in terminal exited after crash report.

### Versions

* **Natron version/commit** 
  This software was compiled from the source code branch tags/2.2.1 at version 35e7f95
   using GNU C++ version 4.8.5 with OpenMP 201107 targetting Intel x86-64 for Linux.

* **OS version:** Debian/stable

**Comment by devernay** [comment]
I think the DiskCache node is independent of the Disk Cache Size (which is for the node cache) and the Viewer Cache Size. DiskCache will just write everything to disk, so use a disk with enough space.
@MrKepzie can you please confirm and update the doc of the DiskCache node accordingly?

**Comment by Ichthyostega** [comment]
Hello,
fist of, I do not want to sound rude. This report was the result of a dozend or so crashes (until I realised the reason), which caused me some loss of work, because in one case it crashed while writing the auto backup. (Fortunately I check in my Natron file regularily into Git, so I lost only about half an hour of work or so).

So the main point is that the application does not handle out-of-diskspace properly, which is a must.

Regarding the cache configuration, there is a preferences page, which allows to set a limit
- for RAM cache size
- "Maximum playback disk cache size"
- "Maximum DiskCache node disk usage"
Plus it allows to set a (single) "disk cache path"

In my case, I set that path to my SSD, which had 15 GiB free. And I set both of the disk cache settings to 5GiB each. It is not clear, if both settings are cumulative, or are meant as a partitioning of a common space, but anyway, at maximum this setting can mean that Natron is allowed to use 10 GiB, including the DiskCache node.

So basically, we have three issues here
- crash on out-of-diskspace in the middle of writing to cache
- not regarding the limits set in the configuration
- lack of ability to set the DiskCache location to a different place than the playback disk cache, because, obviously both have different speed requirements.

**Comment by MrKepzie** [comment]
DiskCache and Playback cache in Natron 2 share the same disk. 
In Natron 2, there are 3 different caches:
- Ram cache
- Disk cache for the viewer
- Disk cache for the DiskCache node
All caches have an independent management of memory and size. 
The 2 disk caches all point to the location you set to "disk cache path". 

We will attempt to reproduce those bugs however we are focusing on Natron 3 now and the Cache system has been completely revamped, so fixing those issues would anyway only be relevant for the current release.

**Comment by Ichthyostega** [comment]
need to add one maybe important detail. On this incident, I previously had activated the "Precache" button in the DiskCache node. This started a job, which, no surprise, caused a lot of load, so after some minutes, I just hit "pause" on that job. And I let it as it was. Maybe this is what lead to the situation causing the crash.

After I realised that I had run out of disk space on the SSD, I removed that paused job and disabled the DiskCache node, and I manually deleted all files from the Cache Dir and then restarted Natron. From that point on, I did not get any crashes. Later, I even re-enabled the DiskCache node, but this did not give me any perceptible speed boost, and it still graphically showed the nodes in the huge network meant to be covered by the disk cache as being pulled. Right now I see

    (hiv)~-$ LC_ALL=C du -hc .cache/Natron/ --max-depth=1
    1.1M	.cache/Natron/DiskCache
    5.5G	.cache/Natron/ViewerCache
    5.5G	.cache/Natron/
    5.5G	total

...which is not entirely correct either (the `du` utility displays in powers of 1024), but it is at least in the correct order of magnitude.

**Comment by Ichthyostega** [comment]
Hello MrKepzie, thanks for the response....
Personally, I am aware of the issue now and know to work around it. IMHO, you should concentrate your (always) limited resources to Natron 3 and get the caching to work as expected there. Maybe add a warning to the documentation for now... :-D

**Comment by devernay** [comment]
This issue was moved to NatronGitHub/Natron#192
